R/utils.R

```diff
@@ -142,3 +142,23 @@ is_windows <- function() {
 }
 
 compact <- function(x) x[!vapply(x, is.null, logical(1))]
+
+set_odbcsysini <- function() {
+  odbcsysini <- Sys.getenv("ODBCSYSINI")
+
+  if (!identical(odbcsysini, "")) {
+    return(invisible(TRUE))
+  }
+
+  odbcsysini <- tryCatch(odbcListConfig(), error = function(e) e)
+
+  if (inherits(odbcsysini, "error")) {
+    return(invisible(TRUE))
+  }
+
+  odbcsysini <- dirname(odbcsysini[1])
```

Comment:
I think I'd write this like:

```suggestion
  tryCatch(
    {
      path <- dirname(odbcListConfig()$drivers)
      Sys.setenv(ODBCSYSINI = path
    }, 
    error = function(err) NULL
  )
  
  invisible()
```

And then change the return above too.

Reply:
You got it--[7a62336](https://github.com/r-dbi/odbc/pull/733/commits/7a623367e7b6fc14a5586d41630ef9fde1b67bc6).